usaspending_api/accounts/views/federal_accounts_v2.py

```diff
@@ -447,13 +450,22 @@ def post(self, request, format=None):
                          'treasuryappropriationaccount__account_balances__total_budgetary_resources_amount_cpe'),
                      managing_agency=Subquery(agency_subquery.values('name')[:1]),
                      managing_agency_acronym=Subquery(agency_subquery.values('abbreviation')[:1]))
-
         # add keyword filter, if it exists
         if keyword:
             queryset = queryset.filter(Q(account_name__icontains=keyword) |
                                        Q(account_number__contains=keyword) |
                                        Q(managing_agency__icontains=keyword) |
                                        Q(managing_agency_acronym__contains=keyword.upper()))
+        if agency_id:
+            if agency_id == DOD_CGAC:
+                tta_list = DOD_ARMED_FORCES_CGAC
+            else:
+                tta_list = agency_id
+
+            tta_filter = Q()
+            for tta in tta_list:
```

Comment:
There might be a problem when looping since `agency_id` is a string when provided by the API and  is not `DOD_CGAC`.

If `DOD_ARMED_FORCES_CGAC` is a list then `tta_list` needs to always be a list